keps/sig-apps/2214-indexed-job/README.md

```diff
@@ -424,64 +424,60 @@ _This section must be completed when targeting beta graduation to a release._
 
 * **What specific metrics should inform a rollback?**
 
+  - job_sync_duration_seconds shows significantly more latency for Indexed Jobs.
+  - job_sync_total shows errors for Indexed Jobs only.
+  - job_finished_total shows that Indexed Jobs don't finish.
+
 * **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
-  Describe manual testing that was done and the outcomes.
-  Longer term, we may want to require automated upgrade/rollback tests, but we
-  are missing a bunch of machinery and tooling and can't do that now.
+
+  Manual test:
+  
+  1. Deploy k8s 1.21 cluster
+  1. Upgrade to 1.22
+  1. Create Indexed Job with big number of completions and pods that run for ~10min.
+  1. Downgrade to 1.21. Verify that no new pods are created for the Indexed Job.
+  1. Upgrade to 1.22. Verify that new pods are created for Indexed Job.
 
 * **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
 fields of API types, flags, etc.?**
-  Even if applying deprecation policies, they may still surprise some users.
+
+  No
 
 ### Monitoring Requirements
 
 _This section must be completed when targeting beta graduation to a release._
 
 * **How can an operator determine if the feature is in use by workloads?**
-  Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
-  checking if there are objects with field X set) may be a last resort. Avoid
-  logs or events for this purpose.
+
+  - job_sync_total has values for the label completion_mode=Indexed.
 
 * **What are the SLIs (Service Level Indicators) an operator can use to determine 
 the health of the service?**
-  - [ ] Metrics
-    - Metric name:
-    - [Optional] Aggregation method:
-    - Components exposing the metric:
-  - [ ] Other (treat as last resort)
-    - Details:
+
+  - [x] Metrics
+    - Metric name (all new): job_sync_duration_seconds, job_sync_total, job_finished_total
+    - Components exposing the metric: kube-controller-manager
 
 * **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
-  At a high level, this usually will be in the form of "high percentile of SLI
-  per day <= X". It's impossible to provide comprehensive guidance, but at the very
-  high level (needs more precise definitions) those may be things like:
-  - per-day percentage of API calls finishing with 5XX errors <= 1%
-  - 99% percentile over day of absolute value from (job creation time minus expected
-    job creation time) for cron job <= 10%
-  - 99,9% of /health requests per day finish with 200 code
+
+  - per-day percentage of job_sync_total with label result=error <= 1%
```

Comment:
I'd like to have clear information what will be counted as an error towards this SLO? For example, I wouldn't include update conflicts as errors. Maybe it's reasonable to either explain what is an error for this metric?

Reply:
I was just thinking of whether the `syncJob` method returns error. But we can be more restrictive. We can discuss in SIG meeting.

Reply:
We didn't get a chance to discuss this in the meeting.

I'm thinking that this metric is a tool for cluster operators to quickly see if things as working as expected in dashboards and easily set alerts.

Let's say there is a rogue controller that is updating job pods, causing conflicts all the time. If we silence Conflict errors, an operator might think that the cluster is operating normally, as no alerts are triggered.

We should not think of this metric as "there is a bug in the controller" (although it can also be used for that), but about making sure that the controller is progressing and there are no agents degrading its operations.